tensorflow/contrib/learn/python/learn/datasets/mnist.py

```diff
@@ -28,7 +28,7 @@
 from tensorflow.python.framework import dtypes
 
 SOURCE_URL = 'http://yann.lecun.com/exdb/mnist/'
-
+BACKUP_URL = 'https://web.archive.org/web/20160828233817/http://yann.lecun.com/exdb/mnist/'
```

Comment:
Instead of this, can we use a more general approach, namely a list of URLs, such as 

```
SOURCE_URLS = ['http://yann.lecun.com/exdb/mnist/', 'https://web.archive.org/web/20160828233817/http://yann.lecun.com/exdb/mnist/']
```

and let the code try them one by one until the first successful one? If all failed, throw an exception.